### DISCUSSION TITLE
[Proposal] Extensions

### DISCUSSION THREAD

**Discussion by Nufflee** [dotnet/csharplang#1010] [General]
I would like to see static extension methods in one of new upcoming C# versions. I'm actually trying to modify Roslyn and implement them on my own. But the one thing that I'm not sure about, how would the syntax look? All the extension methods already need to be static and inside of a static class. So I'll need some help from you guys on this one. Also, quote from an unofficial [C# 8 feature overview](https://stackify.com/csharp-8-features/): 

> Improved Extension Support – Ability to use more than just extension methods. Adding support for properties, static methods and much more.

I will leave this here just so they can take some inspiration from this if this gets more heat.

**Comment by HardlyDifficult** [comment]
Goal:
```csharp
MyClass.CustomExtension(data: 12);
```

Possible implementation:
```csharp
public static class MyClassExtensions 
{
    public static void CustomExtension(
        this MyClass,
        int data)
    {
        ...
    }
}
```

**Comment by Nufflee** [comment]
That looks like a valid approach but we would need a solution for fields, properties, maybe even operators etc. @hardlydifficult

**Comment by iam3yal** [comment]
Check #192.
